source/slang/source-loc.cpp

```diff
@@ -193,159 +370,60 @@ static ExpandedSourceLoc expandSourceLoc(
         }
     }
 
-    // `lo` should now point at the entry we want
-    UInt entryIndex = lo;
-
-    expanded.setRaw(loc.getRaw());
-    expanded.sourceManager = sourceManager;
-    expanded.entryIndex = entryIndex;
-
-    return expanded;
-
-
-}
-
-ExpandedSourceLoc SourceManager::expandSourceLoc(SourceLoc const& loc)
-{
-    return Slang::expandSourceLoc(this, loc);
-}
-
-HumaneSourceLoc getHumaneLoc(ExpandedSourceLoc const& loc)
-{
-    // First check if this location maps to an actual file.
-    SourceFile* sourceFile = loc.getSourceFile();
-    if(!sourceFile)
-        return HumaneSourceLoc();
-
-    auto sourceManager = loc.sourceManager;
-
-    auto& entry = sourceManager->sourceFiles[loc.entryIndex];
-    UInt offset = loc.getRaw() - entry.startLoc.getRaw();
-
-    // We now have a raw input file that we can search for line breaks.
-    // We obviously don't want to do a linear scan over and over, so we will
-    // cache an array of line break locations in the file.
-    auto& lineBreakOffsets = sourceFile->lineBreakOffsets;
-    if( lineBreakOffsets.Count() == 0 )
+    // Check if low is a hit
     {
-        char const* begin = sourceFile->content.begin();
-        char const* end = sourceFile->content.end();
-
-        char const* cursor = begin;
-
-        // Treat the beginning of the file as a line break
-        lineBreakOffsets.Add(0);
-
-        while( cursor != end )
+        SourceUnit* unit = m_sourceUnits[lo];
+        if (unit->getRange().contains(loc))
         {
-            int c = *cursor++;
-            switch( c )
-            {
-            case '\r': case '\n':
-                {
-                    // When we see a line-break character we need
-                    // to record the line break, but we also need
-                    // to deal with the annoying issue of encodings,
-                    // where a multi-byte sequence might encode
-                    // the line break.
-
-                    int d = *cursor;
-                    if( (c^d) == ('\r' ^ '\n'))
-                        cursor++;
-
-                    lineBreakOffsets.Add(cursor - begin);
-                }
-                break;
-
-            default:
-                break;
-            }
+            return unit;
         }
-
-        // Note taht we do *not* treat the end of the file as a line
-        // break, because otherwise we would report errors like
-        // "end of file inside string literal" with a line number
-        // that points at a line that doesn't exist.
     }
 
-    // At this point we can assume the `lineBreakOffsets` array has been filled in.
-    // We will use a binary search to find the line index that contains our
-    // chosen offset.
-    UInt lo = 0;
-    UInt hi = lineBreakOffsets.Count();
+    // Check the parent if there is a parent
+    return (parent) ? parent->findSourceUnit(loc) : nullptr;
```

Comment:
It seems odd to move the lookup in `parent` to the end like this. We can always know whether a range belongs in the parent because the range `[0,N)` always belongs to the parent, so why do the binary search first?